sil-q: init at 1.5.0

kenranunderscore opened the pull request:
###### Description of changes

Adds the [Sil-Q](https://github.com/sil-quirk/sil-q) game, with support for the `ncurses` and `X11` frontends. Sil-Q is the "continuation" of Sil (which is already packaged but no longer actively developed).

The Nix expression takes some things from Sil's and is thus quite similar. One thing I don't know yet is whether I need the `NIX_CFLAGS_COMPILE = "-fcommon"` bit, too. It seems to be relevant to #110571.

This also adds me as a maintainer as a separate commit as per the manual.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

kenranunderscore committed:
maintainers: add kenran

kenranunderscore committed:
sil-q: init at v1.5.0

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

necrophcodr commented on pkgs/games/sil-q/default.nix:
```diff
@@ -0,0 +1,64 @@
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp sil $out/bin
```
You may want to use `install` here with the proper permissions, such as `-m755`.

necrophcodr commented:
I've tried running this using `nix run` and it may be ideal to add `mainProgram = "sil"` as well. Additionally, when running it after building it, it simply produces the following message:

```
./result/bin/sil: Cannot create the '~/.sil/apex/scores.raw' file!

The 'lib' directory is probably missing or broken.
Perhaps the archive was not extracted correctly.
See the manual for more information.
```

Am I missing something here?

necrophcodr commented on pkgs/games/sil-q/default.nix:
```diff
@@ -0,0 +1,64 @@
+    rev = version;
+    sha256 = "sha256-v/sWhPWF9cCKD8N0RHpwzChMM1t9G2yrMDmi1cZxdOs=";
+  };
+  setup = writeScript "setup" ''
```
Is this not being used at all?

kenranunderscore commented:
> I've tried running this using `nix run` and it may be ideal to add `mainProgram = "sil"` as well (please see https://nixos.org/manual/nix/unstable/command-ref/new-cli/nix3-run.html#description for more unformation on this).

Thanks, fixed that! I didn't try it with `nix run` and it's something I'll do in the future for sure.

> Additionally, when running it after building it, it simply produces the following message:
> 
> ```
> ./result/bin/sil: Cannot create the '~/.sil/apex/scores.raw' file!
> 
> The 'lib' directory is probably missing or broken.
> Perhaps the archive was not extracted correctly.
> See the manual for more information.
> ```
> 
> Am I missing something here?

No you're not. I have introduced that by replacing `--run ...` with a `--set ...` in the `wrapProgram` call, and it has worked on my machine because I already had the directory from tests before that, sorry. The `~/.sil` was in single quotes in the wrapper script, so no shell expansion took place. Should be fixed now as well. Thanks for trying it out!

SuperSandro2000 commented on pkgs/games/sil-q/default.nix:
```diff
@@ -0,0 +1,65 @@
+{ lib, stdenv, fetchFromGitHub, writeScript, makeWrapper, ncurses, libX11 }:
+
+let
+  version = "v1.5.0";
+  src = fetchFromGitHub {
+    owner = "sil-quirk";
+    repo = "sil-q";
+    rev = version;
+    sha256 = "sha256-v/sWhPWF9cCKD8N0RHpwzChMM1t9G2yrMDmi1cZxdOs=";
+  };
+  setup = writeScript "setup" ''
+    mkdir -p "$ANGBAND_PATH"
+    # Copy all the data files into place.
+    cp -ar $1/* "$ANGBAND_PATH"
+    # The copied files are not writable, make them so.
+    chmod +w -R "$ANGBAND_PATH"
+  '';
+in stdenv.mkDerivation {
+  pname = "sil-q";
+  inherit version src;
```
```suggestion
  setup = writeScript "setup" ''
    mkdir -p "$ANGBAND_PATH"
    # Copy all the data files into place.
    cp -ar $1/* "$ANGBAND_PATH"
    # The copied files are not writable, make them so.
    chmod +w -R "$ANGBAND_PATH"
  '';
in stdenv.mkDerivation rec {
  pname = "sil-q";
  version = "1.5.0";

  src = fetchFromGitHub {
    owner = "sil-quirk";
    repo = "sil-q";
    rev = "v${version}";
    sha256 = "sha256-v/sWhPWF9cCKD8N0RHpwzChMM1t9G2yrMDmi1cZxdOs=";
  };
```

SuperSandro2000 commented on pkgs/games/sil-q/default.nix:
```diff
@@ -0,0 +1,65 @@
+  # Makefile(s) and config are not top-level.
+  sourceRoot = "source/src";
+
+  prePatch = ''
```
```suggestion
  postPatch = ''
```

kenranunderscore replied:
Will add! What's the reasoning behind this? I guess this applies to the "original" `default.nix` in `sil` as well then.

SuperSandro2000 commented on pkgs/games/sil-q/default.nix:
```diff
@@ -0,0 +1,65 @@
+    # Allow usage of ANGBAND_PATH.
+    substituteInPlace config.h --replace "#define FIXED_PATHS" ""
+
+    # Makefile.std has to be adapted in order to use ncurses.
+    # It says so in its own comments.
```
```suggestion
    # change Makefile.std for ncurses according to its own comment
```

SuperSandro2000 commented on pkgs/games/sil-q/default.nix:
```diff
@@ -0,0 +1,65 @@
+    mkdir -p $out/bin
+    cp sil $out/bin
+    wrapProgram $out/bin/sil \
+      --run "set -u" \
```
```suggestion
```

SuperSandro2000 commented on pkgs/games/sil-q/default.nix:
```diff
@@ -0,0 +1,65 @@
+    cp sil $out/bin
+    wrapProgram $out/bin/sil \
+      --run "set -u" \
+      --run "export ANGBAND_PATH=\$HOME/.sil" \
```
Pleas use the flag to set environment variables instead of shelling.

kenranunderscore replied:
`--set` was what I tried before, but I couldn't get it to work; it kept escaping `$HOME` in some way or other, and the error that was pointed out above was due to me trying to then switch to `~`. I'll try again tomorrow :)

kenranunderscore replied:
I've kept `--run`: It seems impossible to use `--set` and still point to the user's `$HOME` correctly, as the resulting shell code has quotes around the variable value with `--set`. Otherwise the shell output is identical.

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -6684,6 +6684,13 @@
       fingerprint = "932F 3E8E 1C0F 4A98 95D7  B8B8 B0CA A28A 0295 8308";
     }];
   };
+  kenran_ = {
```
```suggestion
  kenran = {
```
I don't think we need the extra underline.
